pennylane/ops/functions/is_commuting.py

```diff
@@ -161,6 +162,36 @@ def commutes_inner(op_name1, op_name2):
 _commutes = _create_commute_function()
 
 
+def _check_opmath_operations(operation1, operation2):
```

Comment:
could potentially  just check `if operation1.pauli_rep is not None`.

Reply:
I agree, this seems like overkill or is there cases that wouldnt be caught with my suggestion [here](https://xanaduhq.slack.com/archives/C06DX049G49/p1709296581611489?thread_ts=1709247345.316049&cid=C06DX049G49)?

Reply:
Aside from the code being simpler it should also be fastest to use the pauli rep of the operators 🙂

Reply:
Thanks both for the suggestion. I didn't know that `.pauli_rep` also works with `sum.` On the other hand, `is_pauli_word` does not recognize sums:

```
p = qml.sum(qml.PauliX(0), qml.prod(qml.PauliY(1), qml.PauliZ(2)))
is_pauli_word(p)
```
results in:

```
False
```

which is why I had to do such a lengthy check. With `.pauli_rep`, the check is much more straightforward.